Modules/_decimal/_decimal.c

```diff
@@ -3663,9 +3663,7 @@ PyDec_AsTuple(PyObject *dec, PyObject *dummy UNUSED)
         }
         else {
             coeff = PyTuple_New(0);
-            if (coeff == NULL) {
```

Comment:
I must say I'm not a fan of this.  Can you use a new static inline function in a header file `PyTuple_EmptyNew()` that cannot fail? And get rid of the assert()?

I don't like checking the return value based on whether it is `PyTuple_New(0)` or `PyTuple_New(1)`.

Reply:
Agreed. I think it is a much safer consistent idiom to document PyTuple_New as always needing its return value checked.

Have a separate new API call to get an empty tuple reference that guarantees the return value never needs to be checked.  `PyTuple_EmptyNew()` as suggested perhaps.  The difference between using an API call and directly referencing `_PyTuple_Empty` is the `Py_INCREF` being done for the caller.

Done this way, the singleton empty tuple can still be created on the heap by PyTuple_New(0) during interpreter initialization rather than being static.